### DISCUSSION TITLE
cross_axis_slope vs cross_axis_tilt

### DISCUSSION THREAD

**Discussion by williamhobbs** [pvlib/pvlib-python#2334] [General]
pvlib.shading.shaded_fraction1d uses '_cross_axis_slope_' as an input, while pvlib.tracking.singleaxis uses '_cross_axis_tilt_'. I think these are the same thing - is that correct? If so, would it make sense to update one of them to be consistent?

I couldn't seem to find any discussion of this in related PRs, but apologies if I just missed it. 

tagging @kandersolar, @mikofski, @AdamRJensen, and @echedey-ls, since you all worked on these functions and/or references.

**Comment by kandersolar** [comment]
Yes, they represent the same quantity.  That they have different names in those two functions was an oversight, and a shame.  Thanks for pointing it out.

I'd have to think a bit more to form an opinion on which of them should be changed to match the other.

**Comment by mikofski** [comment]
I am minus one against “tilt” because the cross axis slope is the angle between adjacent rows. In my opinion tilt refers to the orientation of a tracker axis or module surface.

**Reply by kandersolar** [reply]
"tilt" also tends to be a non-negative quantity in pvlib, whereas `cross_axis_X` can be positive or negative.  

"slope" has the downside of potentially being interpreted as "grade", i.e. rise over run (rather than the arctan of rise over run).

Other places the "tilt" variant appears:
- name of `pvlib.tracking.calc_cross_axis_tilt`
- parameter of `pvlib.pvsystem.SingleAxisTrackerMount`

**Reply by echedey-ls** [reply]
So, shall we change it? I prefer to do that sooner than later, but I don't see any strong opinion (nor I have one). We can even consider `cross_axis_rotation` to address the well-reasoned comment above me.

**Reply by williamhobbs** [reply]
I prefer `cross_axis_slope`. 

> "slope" has the downside of potentially being interpreted as "grade"

Hopefully documentation will be very clear about units. 

> We can even consider `cross_axis_rotation`

I can't help but think of tracker rotation. 

To address the concern about confusing an angle with a grade, we could consider `cross_axis_slope_angle` or `cross_axis_angle`, but I still like `cross_axis_slope` the best.

**Reply by echedey-ls** [reply]
I'm retracting my upvote on Will's comment above mine, see https://github.com/pvlib/pvlib-python/pull/2543#issuecomment-3315277041 for the reasoning and extent of changes deemed necessary.

**Reply by mikofski** [reply]
I’m also changing my vote, despite what I said previously:
>I am minus one against “tilt” 

I think I prefer ”tilt”  for reason explained in https://github.com/pvlib/pvlib-python/pull/2543#issuecomment-3315293966. Although I’m not deeply invested in either sorry for flip-flopping.

**Reply by williamhobbs** [reply]
I'll caveat this with noting that I don't think it matters too much, but I still prefer "slope."

For example, I can mention "slope-aware backtracking," and it is relatively unambiguous what I am talking about. If I were instead to mention "tilt-aware backtracking," I think the meaning is lost. 

And I informally asked a few non-solar folks in the office about the image below. Everyone said the angle of **B** should be "tilt" and **C** should be "slope" or "incline." When I said **B** has to be "tilt" and asked if **A** should be "tilt" or "slope", everyone liked "slope" more and said that "tilt" would be confusing. One person did mention that slope makes them think of "rise over run"...

<img width="967" height="226" alt="image" src="https://github.com/user-attachments/assets/0517607b-4f57-4d8e-ad92-42afd94e4e53" />

**Reply by AdamRJensen** [reply]
I also prefer slope for anything related to the ground, and using tilt for the module surface. And of course consistency, even if it  requires a deprecation cycle.

**Reply by kandersolar** [reply]
> the “tilt” of the plane containing tracker axes

Yep, it's that plane which is tilted.  Note that that plane's orientation is fully defined by `axis_tilt` and `cross_axis_tilt`.  Seems natural to have the terms use parallel terms when you've got all the geometry loaded up in your head, but as @williamhobbs points out, maybe not so much otherwise.

I'm still not sure which way I'd vote.  Abstaining for now :)

**Reply by kurt-rhee** [reply]
> "There are two hard problems in computer science, cache invalidation and naming things" - Phil Karlton

+1 for slope
